lib/step_flow/workflows/workflows.ex

```diff
@@ -191,8 +146,13 @@ defmodule StepFlow.Workflows do
         query
 
       date_value ->
-        date = Date.from_iso8601!(date_value)
-        from(workflow in query, where: fragment("?::date", workflow.inserted_at) <= ^date)
+        date =
+          case NaiveDateTime.from_iso8601(date_value) do
+            {:ok, date} -> date
+            _ -> Date.from_iso8601!(date_value)
+          end
+
+        from(workflow in query, where: fragment("?::timestamp", workflow.inserted_at) <= ^date)
```

Comment:
as you've changed from `date` to `timestamp`, the `Date.from_iso8601` is incorrect. You need to convert into a `DateTime` before